src/modules/chat/assistant-plan/PlanStep.tsx

```diff
@@ -304,6 +309,15 @@ export function PlanStep({ step, toolCall }: Props) {
                         <TraceInfoView data={stepTrace.data} />
                       </motion.section>
                     )}
+                    {debugMode &&
+                      !traceError &&
+                      !stepTrace &&
+                      expanded &&
+                      allStepsDone && (
+                        <motion.section {...fadeProps()} key={`${id}:trace`}>
+                          <Spinner />
+                        </motion.section>
+                      )}
```

Comment:
For `AnimatePresence` to work properly, it should only have one direct descendant, so I think you need to split it into two separate occurrences of `AnimatePresence` here.

I am also not sure about the condition, when to render the loader. You do it only on debugMode on, if step expanded and allStepsDone, but the data itself that are above are shown only if stepTrace present? Is that correct behavior?

Reply:
> I am also not sure about the condition, when to render the loader. You do it only on debugMode on, if step expanded and allStepsDone, but the data itself that are above are shown only if stepTrace present? Is that correct behavior?

Yes, this condition is right. The stepTrace data are loaded only in the `debugMode`, So I don't need to test the debug mode again. `allStepsDone` is only to prevent premature loading.

Reply:
`AnimatePresence` per child => done